### PR TITLE
PD-424: Enable use of individual icon components

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

This pull request adds the ability to directly use Icons, rather than using them through the `Icon` component.

🎟 _Jira ticket:_ [Explore importing individual icon components into leafygreen-ui components](https://jira.mongodb.org/browse/PD-424)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/icon/README.md [hunk 1]
```diff
@@ -88,3 +88,34 @@ const myGlyphs = {
...
+We export each icon individually to enable using them directly. This is particularly useful if you don't intend to use more than a few icons, and should allow bundlers to omit unused Icons more effectively, reducing bundle size.
+
+> **NOTE**
+>
+> This API is only useful if you don't use the default `Icon` component elsewhere in your bundled code. If you intend to use a large portion of the default set of icons, we recommend continuing to use the default `Icon` component for the best developer experience.
+
+## Example
+
+```js
+import { glyphs } from '@leafygreen-ui/icon';
```
Comment (hswolff) [inline packages/icon/README.md]: API should be `<Icon.Plus />`, no? Or `<Glyph.Plus />`. Plural implies a collection, this is an enumeration of items.

**Comment by greggb** [comment]
This still on the radar?

**Comment by DesignerDave** [comment]
@greggb Still on the radar – unfortunately haven't had the time to respond to feedback for a few weeks on this.

**Comment by greggb** [comment]
So excited

**Comment by DesignerDave** [comment]
This project seems to be a lot larger than originally expected. It's still top-of-mind, but this is not the solution. Closing this PR, but expect another on this topic.

**Pull request closed without merging**